src/hotspot/cpu/x86/assembler_x86.inline.hpp

```diff
@@ -29,54 +29,6 @@
 #include "asm/codeBuffer.hpp"
 #include "code/codeCache.hpp"
 
-#ifndef _LP64
-inline int Assembler::prefix_and_encode(int reg_enc, bool byteinst, bool is_map1)
-{
-    int opc_prefix = is_map1 ? 0x0F00 : 0;
-    return opc_prefix | reg_enc;
-}
-
-inline int Assembler::prefixq_and_encode(int reg_enc, bool is_map1) {
-    int opc_prefix = is_map1 ? 0xF00 : 0;
-    return opc_prefix | reg_enc;
-}
-
-inline int Assembler::prefix_and_encode(int dst_enc, bool dst_is_byte, int src_enc, bool src_is_byte, bool is_map1) {
-    int opc_prefix = is_map1 ? 0xF00 : 0;
-    return opc_prefix | (dst_enc << 3 | src_enc);
-}
-
-inline int Assembler::prefixq_and_encode(int dst_enc, int src_enc, bool is_map1) {
-    int opc_prefix = is_map1 ? 0xF00 : 0;
-    return opc_prefix | dst_enc << 3 | src_enc;
-}
-
-inline void Assembler::prefix(Register reg) {}
-inline void Assembler::prefix(Register dst, Register src, Prefix p) {}
-inline void Assembler::prefix(Register dst, Address adr, Prefix p) {}
-
-inline void Assembler::prefix(Address adr, bool is_map1) {
-    if (is_map1) {
-        emit_int8(0x0F);
-    }
-}
-
-inline void Assembler::prefixq(Address adr) {}
-
-inline void Assembler::prefix(Address adr, Register reg,  bool byteinst, bool is_map1) {
-    if (is_map1) {
-        emit_int8(0x0F);
-    }
-}
-inline void Assembler::prefixq(Address adr, Register reg, bool is_map1) {
-    if (is_map1) {
-        emit_int8(0x0F);
-    }
-}
-
-inline void Assembler::prefix(Address adr, XMMRegister reg) {}
-inline void Assembler::prefixq(Address adr, XMMRegister reg) {}
-
-#endif // _LP64
+// TODO: Remove?
```

Comment:
Yes, remove. There are no files which include it anymore.

Reply:
Ah no, here is the use:

```
/home/shade/trunks/jdk/src/hotspot/share/asm/assembler.inline.hpp:30:1: fatal error: assembler_x86.inline.hpp: No such file or directory
   30 | #include CPU_HEADER_INLINE(assembler)
      | ^~~~~~~~~~~~~
```

And we have to leave this, as some platforms do have non-empty `.inline.hpp`-s. I'll just remove the TODO.

Reply:
Also transitive includes, grr. I pushed new commit that should handle all of this.